### PR TITLE
Fix broken tests and VS solutions caused by source code reorganization and other build infra cleanup

### PR DESCRIPTION
**Pull request by natemcmaster**
Changes:
* Fix broken tests and VS solutions caused by source code reorganization
* Add a check to validate generated code and solutions on PRs (resolves https://github.com/aspnet/AspNetCore-Internal/issues/1611 - @Tratcher @JamesNK )
* backport some source code reorg to src/Identity
* Fix startvs.cmd if you've already run build.ps1
* Add PR checks for tests on Linux/macOS
* Skip broken Nginx tests
* Add conditions to skip IIS tests on non-Windows platforms

### CODE REVIEW

**Comment by jkotalik** [comment]
Can you add the following change to IISIntegration too: https://github.com/aspnet/AspNetCore/pull/6313/files#diff-bcc148bf5601e2bfc4599adaa48570a5R23. This will fix those tests.

**Comment by adrian109** [comment]
The AspNetCore-pr-validation-temp pipeline build appears to have 3 build tasks and 1 code check task.  It appears that they run independently from each other.  Shouldn't the code check task be a prerequisite for the build tasks. i.e. the builds only execute if the code checks were successful?

**Comment by natemcmaster** [comment]
> Shouldn't the code check task be a prerequisite for the build tasks. i.e. the builds only execute if the code checks were successful?

I don't think it has to be. The benefit of treating them as separate PR checks is that you can still get build and test validation on whether or not codechecks pass. Some code checks may not cause the build to actually fail and are just about keeping code clean.

**Comment by adrian109** [comment]
> [3fcbf36](https://github.com/aspnet/AspNetCore/commit/3fcbf36727f7194385cfb88c5bb579e558d4d07c)

Line 23 is missing a comma at the end

**Comment by jkotalik** [comment]
Seems like skip conditions for IIS aren't working in 2.1 for mac. Also nginx tests are failing.

There was a few OutOfMemoryExceptions from some Identity tests too. 😢

File: eng/scripts/CodeCheck.ps1 [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            | % {
+                $proj = Join-Path $slnDir $_
+                if (-not (Test-Path $proj)) {
+                    $errors += "Missing project. Solution references a project which does not exist: $proj. [$sln] "
+                }
+            }
+        }
+
+    #
+    # Generated code check
```
Comment (JamesNK) [inline eng/scripts/CodeCheck.ps1]: Good idea 👍

File: eng/scripts/common.psm1 [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    if ($RemotePath -notlike 'http*') {
+        Copy-Item $RemotePath $LocalPath
+        return
+    }
+
+    $retries = 10
+    while ($retries -gt 0) {
+        $retries -= 1
+        try {
+            Invoke-WebRequest -UseBasicParsing -Uri $RemotePath -OutFile $LocalPath
```
Comment (JamesNK) [inline eng/scripts/common.psm1]: You could add an optional retry count parameter to Invoke-Block and reuse it down here.

I have used something similar with success in build scripts to make operations more reliable - https://github.com/JamesNK/Newtonsoft.Json/blob/master/Build/build.ps1#L291-L311

File: eng/scripts/CodeCheck.ps1 (hunk 1 above)
Comment (natemcmaster) [reply eng/scripts/CodeCheck.ps1]: This is just a start. I'm hoping to add more checks in the future, like updating resx files, generated baselines, and maybe even a C# code formatter.

**Comment by natemcmaster** [comment]
I'm going to completely disable some Identity tests in PR builds. Let's use https://github.com/aspnet/AspNetCore/issues/5812 to track re-enabling. Identity tests never worked in Azure Pipelines before we merged the source code into one repo, and I expect we're seeing the same problems now.

**Comment by natemcmaster** [comment]
Progress! Still not to 100% green, but I'd like to merge anyways since this fixes a few hundred tests.

Remaining test failures:

Windows:
* Two failures in Razor tests. https://github.com/aspnet/AspNetCore-Internal/issues/1614

macOS:
* Nginx tests. Already known issue, tracked by https://github.com/aspnet/AspNetCore-Internal/issues/1525
* IIS tests are still running. That's weird. Any ideas why? @pakrym @jkotalik 

Linux:
* Something is borked with AzDO agents right now. Not sure how to fix this one: 
image

**Comment by Tratcher** [comment]
The nginx tests were disabled in master by https://github.com/aspnet/AspNetCore/pull/6062. Feel free to backport.

**Comment by natemcmaster** [comment]
:up: :date:

Skipping Nginx tests and IIS functional tests on non-Windows.

**Comment by jkotalik** [comment]
Add an OSSkipCondition here: https://github.com/aspnet/AspNetCore/blob/release/2.1/src/Servers/IIS/IISIntegration/test/FunctionalTests/Properties/AssemblyInfo.cs#L7

**Comment by jkotalik** [comment]
What you did also works.

**Comment by natemcmaster** [comment]
Good suggestion. Made the skip condition apply to the whole assembly instead of duplicating this on all test classes.

**Review by jkotalik — APPROVED** [review]

**Comment by natemcmaster** [comment]
All but 2 tests pass :tada: on Windows. MacOs and Linux tests pass. Let's use https://github.com/aspnet/AspNetCore-Internal/issues/1614 to follow up on those.

**Pull request merged**